ext/session/session.c

```diff
@@ -873,7 +873,7 @@ PS_SERIALIZER_DECODE_FUNC(php_serialize) /* {{{ */
 	Z_ADDREF_P(&PS(http_session_vars));
 	zend_hash_update_ind(&EG(symbol_table), var_name, &PS(http_session_vars));
 	zend_string_release_ex(var_name, 0);
-	return SUCCESS;
+	return result || !vallen ? SUCCESS : FAILURE;
```

Comment:
Why is it necessary to special-case `!vallen`?

Reply:
Unserializing an empty string returns 0, but an empty session string must be allowed. We could catch an empty string earlier (and not call `php_var_unserialize()` at all).